### PR TITLE
chore(docs): Supabase integration docs, and Checkout example

### PR DESCRIPTION
**Pull request by samwillis**

### CODE REVIEW

File: docs/examples/checkout.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# Checkout example
+
+<div className="sm:float-right sm:w-2/5">
+
+  [![Store and checkout example screenshot](./checkout-screenshot.png)](https://checkout-demo.electric-sql.com)
+
+</div>
+
+This is an example of an online store and checkout built using Electric. It additionally uses the Supabase suite of tools to both host the Postgres database, as well as using Supabase Auth, and Edge Functions. Finally, the UI is built with the Ionic framework.
```
[the remark this answers is not in the document]
Comment (thruflo) [reply docs/examples/checkout.md:18]: Ditto direct links for database, auth and functions.

File: docs/integrations/deployment/supabase.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+---
+title: Supabase
+description: >-
+  An open source Firebase alternative built on Postgres.
+sidebar_position: 30
+---
+
+ElectricSQL supports connecting to a hosted Postgres provided by [Supabase](https://supabase.com), an open source Firebase alternative that also provides many other tools, including Authentication and Edge Functions. All Supabase Postgres instances already have logical replication enabled, and so connecting is super easy.
+
+### Deploying Electric with Supabase
```
[the remark this answers is not in the document]
Comment (thruflo) [reply docs/integrations/deployment/supabase.md:10]: I would also signpost the steps of the instructions with a bullet point list of section links.

File: docs/integrations/deployment/supabase.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+:::info
+Supabase support for Electric is currently only enabled in the canary build, available as the `electricsql/electric:canary` docker image.
+:::
+
+First, retrieve the connection details for your database by going to "Project Settings" > "Database". The top of the screen will list the host, database name, port, and user for your Postgres database. Your password for the database will have been set when you created the project. Use this to construct your `DATABASE_URL` in for the form:
```
Comment (thruflo) [inline docs/integrations/deployment/supabase.md:16]: Here I think you need to step back and say first thing is to create a hosted Postgres database on Supabase. To do that, go here. Click on this and follow the instructions here.

Then once you have a Supabase database, go to the project Settings > Database page in the Supabase dashboard ...

File: docs/integrations/deployment/supabase.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+```
+postgresql://user-name:password@db.your-host.supabase.co:port/database-name
+```
+
+Next, you must configure Electric to use the `direct_writes` mode for inbound transactions, as Supabase does not support the default inbound `logical_replication` mode. This is configured with the `ELECTRIC_INBOUND_MODE` environment variable.
```
Comment (thruflo) [inline docs/integrations/deployment/supabase.md:22]: Here we've jumped into configuring Electric without explaining that you need to deploy it. And where to deploy it.

File: docs/integrations/deployment/supabase.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    electricsql/electric:canary
+```
+
+See the [full configuration options](../../api/service.md#configuration-options) for more details.
+
+:::caution
+Do not use the "Connection Pool" connection string that Supabase provides for your database, as this will prevent the sync service from operating. 
+:::
+
+### Using Supabase Auth with Electric
```
Comment (thruflo) [inline docs/integrations/deployment/supabase.md:57]: We've jumped here into other Supabase integrations. Without explaining how to verify the Electric connection. And without running migrations.

I would put the supabase integration subsections below a shared heading.

File: docs/usage/installation/postgres.md [hunk 6]
```diff
@@ -19,6 +19,7 @@ This includes, for example:
 
 - [AWS RDS](https://repost.aws/knowledge-center/rds-postgresql-use-logical-replication) and [Aurora](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Replication.Logical.html) (including Aurora Serverless v2)
 - [Crunchy Data](https://www.crunchydata.com) who have a free tier and logical replication enabled by default
+- [Supabase](https://supabase.com/) an open source Firebase alternative build on Postgres - see out [integration guide](../../integrations/deployment/supabase.md)
```
Comment (thruflo) [inline docs/usage/installation/postgres.md:22]: I would suggest:

```mdx
- [Supabase](https://supabase.com/) who provide managed Postgres hosting with logical replication and a suite of other backend-as-a-service tools, including auth and edge functions (see <DocPageLink path="integrations/deployment/supabase" /> for more information about using Electric with Supabase)
```

File: docs/examples/checkout.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+## Supabase integration
+
+To build this demo we used the [Supabase suite of tools](https://supabase.com/), both to host the Postgres database, as well as using Supabase Auth, and Edge Functions. Finally, the UI is built with the Ionic framework.
```
Comment (alco) [inline docs/examples/checkout.md:64]: This sentence reads rather awkwardly. Maybe rephrase to something like "The example app's UI is built with the Ionic framework and Supabase's suite of tools, namely, the hosted Postgres database, Supabase Auth, and Edge Functions".

File: docs/examples/checkout.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+Authentication for the store uses [Supabase Auth](https://supabase.com/docs/guides/auth), using their standard form component, and then uses the same JWT token for authenticating with the Electric sync service.
+
+A [Supabase Edge Function](https://supabase.com/docs/guides/functions) is used to process orders when they are placed. This is a called from an "on insert" trigger on the 'orders table' in Postgres. When an order is placed, the order is entered as a row into the local orders table in the app; this is then synchronised with the Postgres table.
```
Comment (alco) [inline docs/examples/checkout.md:68]: On the first reading, I was lost a bit after reading about the "trigger on the orders table in Postgres". The phrase "when an order is placed" in the next sentence after it didn't register in my mind as occurring in the app, I was still mentally attached to Postgres in that short time span.

I've tried to move the words around to make it clearer.

```suggestion
A [Supabase Edge Function](https://supabase.com/docs/guides/functions) is used to process orders when they are placed. This is a called from an "on insert" trigger on the 'orders table' in Postgres. When an order is placed in the app, it is entered as a row into the local orders table; this is then synchronised with the remote Postgres table by Electric.
```

File: docs/integrations/deployment/supabase.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+:::caution
+Do not use the "Connection Pool" connection string a little further down the screen that Supabase provides for your database, as this will prevent the sync service from operating. 
+:::
+
+### 3. Configuring Electric to connect to Supabase
+
+Next, we configure Electric to connect to Supabase. The [Electric sync service](../../api/service) is available as either a Docker image or Elixir app, and it uses environment variables for configuration.
+
+You must configure Electric to use the `direct_writes` mode for inbound transactions, as Supabase does not support the default inbound `logical_replication` mode. This is configured with the `ELECTRIC_INBOUND_MODE` environment variable.
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:50]: The setting has been renamed in [my PR](https://github.com/electric-sql/electric/pull/698) (but not yet in the canary Docker image).

```suggestion
You must configure Electric to use the `direct_writes` mode for inbound transactions, as Supabase does not support the default inbound `logical_replication` mode. This is configured with the `ELECTRIC_WRITE_TO_PG_MODE` environment variable.
```

File: docs/integrations/deployment/supabase.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+Along with the other [configuration options](../../api/service.md#configuration-options), these details will result in an environment configuration for Electric, similar to:
+
+```bash
+DATABASE_URL=postgresql://user-name:password@db.your-host.supabase.co:port/database-name
+ELECTRIC_INBOUND_MODE=direct_writes
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:56]:
```suggestion
ELECTRIC_WRITE_TO_PG_MODE=direct_writes
```

File: docs/integrations/deployment/supabase.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+Depending on how you run Electric these could be passed as arguments to Docker, set in a ENV file or entered into a managed host's dashboard.
+
+### 4. Running a local Electric Docker connected to Supabase
+
+To run a local Electric Docker and connect it to your Supabase Postgres you can run the following command.
+
+```bash
+docker run \
+    -e "DATABASE_URL=postgresql://user-name:password@db.your-host.supabase.co:port/database-name" \
+    -e "ELECTRIC_INBOUND_MODE=direct_writes" \
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:71]:
```suggestion
    -e "ELECTRIC_WRITE_TO_PG_MODE=direct_writes" \
```

File: docs/integrations/deployment/supabase.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+
+For details on how to run migrations see our [migrations documentation](../../usage/data-modelling/migrations.md).
+
+:::caution
+It's important not to make schema changes to your database via the Supabase dashboard after a table has been "Electrified" as these changes will not be tracked by Electric and propagated to client databases.
+:::
+
+## Using other Supabase tools with Electric
+
+Supabase provide a suite of tools that pair well with Electric when building local-first apps, these include [Supabase Auth](#supabase-auth) and [Supabase Edge Functions](#supabase-edge-functions).
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:99]:
```suggestion
Supabase provides a suite of tools that pair well with Electric when building local-first apps, these include [Supabase Auth](#supabase-auth) and [Supabase Edge Functions](#supabase-edge-functions).
```

File: docs/integrations/deployment/supabase.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+EXECUTE FUNCTION call_my_edge_function_trigger();
+
+-- This next line is required to ensure that triggers on the table are called
+-- as a result of the Electric sync
+ALTER TABLE "public"."my_table" ENABLE ALWAYS TRIGGER my_edge_function_trigger;
+```
+
+This code can either be run directly against your Postgres via the Supabase console, or you can include it in your database migrations.
+
+You can see an example of this pattern in our [Checkout Example](https://github.com/electric-sql/electric/blob/main/examples/checkout/)
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:179]:
```suggestion
You can see an example of this pattern in our [Checkout Example](https://github.com/electric-sql/electric/blob/main/examples/checkout/).
```

File: docs/integrations/deployment/supabase.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+    -p 5133:5133 \
+    -p 65432:65432 \
+    electricsql/electric:canary
+```
+
+This will start Electric in [insecure mode](../../api/service.md#authentication) and connect it to your database, which is perfect for local development. The logs will be printed to the terminal allowing you to see any errors that may occur.
+
+### 5. Verifying that Electric has successfully connected
+
+Once you have the sync service running, it's time to verify that it has successfully connected to Supabase. The easiest way to do this is via the Supabase dashboard.
```
Comment (alco) [inline docs/integrations/deployment/supabase.md:83]: The actual "easiest" way is to run `curl http://<electric host>:5133/api/status` and check if it returns `Connection to Postgres is up!`.

What you're describing in this section looks more like "Verifying that Electric has successfully initialized its Supabase schema" to me.

Comment (samwillis) [reply docs/integrations/deployment/supabase.md:83]: Good call! It doesn't look like this is documented anywhere in our docs.

File: docs/integrations/deployment/supabase.md [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+
+First, if you don't yet have a Supabase account visit [supabase.com](https://supabase.com) and create one.
+
+Creating a Postgres database with Supabase is easy. First, log in to the dashboard and click "New Project". In the form enter a name for the database, and a password that will be used to connect to it. Make a note of this password as you will need it when connecting Electric to Supabase.
+
+You will also need to select a region for you database to be hosted in. It's recommended to host both your Postgres and Electric sync service in the same region if possible to reduce latency.
+
+### 2. Retrieving the connection details
+
+You can retrieve the connection details for your database by going to "Project Settings" > "Database" in the Supabase dashboard. The top of the screen will list the `host`, `database name`, `port`, and `user` for your Postgres database. Your password for the database will have been set when you created the project. Use this to construct your `DATABASE_URL` in the form of:
```
[the remark this answers is not in the document]
Comment (samwillis) [reply docs/integrations/deployment/supabase.md:36]: Thanks!

**Review by thruflo — APPROVED** [review]

File: docs/integrations/deployment/supabase.md (hunk 9 above)
Comment (alco) [reply docs/integrations/deployment/supabase.md]: @thruflo pushed a new image and updated the variable name in the docs.
